core/executor/src/precompiles/modexp.rs

```diff
@@ -24,14 +24,13 @@ impl PrecompileContract for ModExp {
         _context: &Context,
         _is_static: bool,
     ) -> Result<(PrecompileOutput, u64), PrecompileFailure> {
-        let large_number = LargeNumber::parse(input)?;
-
         let gas = Self::gas_cost(input);
         if let Some(limit) = gas_limit {
             if limit < gas {
                 return err!();
             }
         }
+        let large_number = LargeNumber::parse(input)?;
```

Comment:
Other implementations handle `base = 0` without reading the input, we should probably do the same.

Reply:
no, we only support `eip_2565_cost`

> handle base_len = 0 or mod_len = 0

is not suitable eip 2565

Reply:
`multiplication_complexity` is still zero when both `base_size` and `modulo_size` are zero.

Reply:
yeah, I add these codes to quick return
```rust
      // multiplication_complexity always zero
      if base_size == 0 && modulo_size == 0 {
          return Self::MIN_GAS;
      }
```

Reply:
The point is we shouldn't read `e` in this case because `exponent_size` may be large but gas price is small.

Reply:
done

Reply:
It should be handled before `LargeNumber::parse`, which will read the exponent. E.g. https://github.com/ethereum/go-ethereum/blob/a03490c6b2ff0e1d9a1274afdbe087a695d533eb/core/vm/contracts.go#L374-L377

Reply:
This is about execution, not gas calculation.

Reply:
There is such a condition on the execution side, but it was originally necessary to return after the full analysis is completed, and now it is advanced before the analysis